### PR TITLE
doc: Update URLs to HTTPS where necessary 

### PR DESCRIPTION
**Pull request by sbilling**
Addresses #20807  - updating URLs from HTTP to HTTPS where necessary. Due to the age of the Bitcoin codebase there are a number of legacy URLs with `http://` that support `https://` - I used a script to locate all of them and then visited each one to ensure they are available behind a secure domain. A copy of the script I used can be found on the issue #20807

### CODE REVIEW

**Review by stevenans9859 — APPROVED** [review]

**Review by stevenans9859 — APPROVED** [review]

**Review by stevenans9859 — APPROVED** [review]

File: src/univalue/configure.ac [hunk 1]
```diff
@@ -33,7 +33,7 @@ LIBUNIVALUE_MICRO_VERSION=libunivalue_micro_version
 LIBUNIVALUE_INTERFACE_AGE=libunivalue_interface_age
 
 # ABI version
-# http://www.gnu.org/software/libtool/manual/html_node/Updating-version-info.html
+# https://www.gnu.org/software/libtool/manual/html_node/Updating-version-info.html
```
Comment (MarcoFalke) [inline src/univalue/configure.ac:36]: This is a subtree that needs to be updated upstream

File: src/leveldb/doc/benchmark.html [hunk 2]
```diff
@@ -451,7 +451,7 @@ <h4>Random Reads</h4>
...
-<p>The preceding numbers are for an ext3 file system. Synchronous writes are much slower under <a href="http://en.wikipedia.org/wiki/Ext4">ext4</a> (LevelDB drops to ~31 writes / second and TreeDB drops to ~5 writes / second; SQLite3's synchronous writes do not noticeably drop) due to ext4's different handling of <span class="code">fsync</span> / <span class="code">msync</span> calls. Even LevelDB's asynchronous write performance drops somewhat since it spreads its storage across multiple files and issues <span class="code">fsync</span> calls when switching to a new file.</p>
+<p>The preceding numbers are for an ext3 file system. Synchronous writes are much slower under <a href="https://en.wikipedia.org/wiki/Ext4">ext4</a> (LevelDB drops to ~31 writes / second and TreeDB drops to ~5 writes / second; SQLite3's synchronous writes do not noticeably drop) due to ext4's different handling of <span class="code">fsync</span> / <span class="code">msync</span> calls. Even LevelDB's asynchronous write performance drops somewhat since it spreads its storage across multiple files and issues <span class="code">fsync</span> calls when switching to a new file.</p>
```
Comment (MarcoFalke) [inline src/leveldb/doc/benchmark.html:454]: same

**Comment by practicalswift** [comment]
Concept ACK, but please squash the commits and remove the changes to the following files/directories:
* `contrib/macdeploy/macdeployqtplus`: seems to be unrelated whitespace changes
* `depends/patches/`: modifying patches will break them
* `src/leveldb/`: this is a dependency and change should be submitted upstream
* `src/univalue/`: this is a dependency and change should be submitted upstream

Thanks for addressing this! :)

**Comment by theStack** [comment]
Concept ACK
Happy to review as soon as all previous review comments are addressed.

**Comment by fanquake** [comment]
Thanks for your contribution. However to move things along I'm going to fix this up and consolidate it with some other changes. Your commit will still be attributed to you.

**Pull request closed without merging**